src/utils/get-config.js

```diff
@@ -8,13 +8,31 @@ import CONSTANTS from '../constants';
 const PROTOCOL_RE = /^([\w-]+?)(?=\:\/\/)/;
 
 export default function (id, mainPath) {
-    if (statSync(mainPath).isDirectory())
-        mainPath = path.join(mainPath, CONSTANTS.configFileName);
+    let mainPathJs = mainPath;
+    let mainPathJson = mainPath;
 
-    var mainDir      = path.dirname(mainPath);
-    var configString = readFileSync(mainPath).toString();
+    if (statSync(mainPath).isDirectory()) {
+        mainPathJs = path.join(mainPath, CONSTANTS.configFileName + '.js' );
+        mainPathJson = path.join(mainPath, CONSTANTS.configFileName );
+    }
+
+    var mainDir      = path.dirname(mainPathJs);
+
+    var config;
+
+    try {
+        config = require(mainPathJs);
+    }
+    catch ( e ) {
 
-    var config = JSON.parse(configString);
+        if ( !e.message.includes('Cannot find module') )
+
+            throw e;
```

Comment:
I'm still not fond of masking the `Cannot find module` issue. The first problem is that the error message can be changed without a notice in the next Node.js version. The second is that you can specify a full path to a JS config and make a mistake in it. If we mask the error, it can be very hard to detect a misprint without a proper error message.

If you don't want to touch `require.extensions`, we can define a constant list that contains `.js` and `.json` extensions (`require` can load JSON files too), and use `require` if we can detect a supported extension in a config path, and fallback to the legacy loading mechanism otherwise.

Reply:
> The first problem is that the error message can be changed without a notice in the next Node.js version
This seems at odds with the dependency on code marked as deprecated to me.
>The second is that you can specify a full path to a JS config and make a mistake in it
Good point.
>and use require if we can detect a supported extension in a config path, and fallback to the legacy loading mechanism otherwise.
This sounds like best of both worlds 👍